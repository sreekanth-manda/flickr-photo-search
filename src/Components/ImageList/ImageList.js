import React from "react";
import shortid from 'shortid';
import Image from '../Image/Image';
import './ImageList.scss';

const ImageList = ({ images }) => (
  <div className="image-list">
    {images.map((image, idx) => <Image key={shortid.generate()} image={image} idx={idx} />)}
  </div>
);

export default ImageList;
