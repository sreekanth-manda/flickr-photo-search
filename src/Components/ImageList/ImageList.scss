.image-list {
  display: flex;
  flex-wrap: wrap;
  flex-direction: row;
}