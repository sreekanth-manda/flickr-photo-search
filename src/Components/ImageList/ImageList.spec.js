import React from 'react';
import { mount } from 'enzyme';
import ImageList from './ImageList';

describe('ImageList', () => {
  const compoentProps = {
    images: [
      {
        farm: 66,
        id: "48458406181",
        isfamily: 0,
        isfriend: 0,
        ispublic: 1,
        owner: "46922592@N00",
        secret: "23a99e54c0",
        server: "65535",
        title: "blah title 1",
      },
      {
        farm: 68,
        id: "232323123",
        isfamily: 0,
        isfriend: 0,
        ispublic: 1,
        owner: "46922592@N00",
        secret: "23a99e54c0",
        server: "65535",
        title: "blah title 2",
      }
    ]
  }

  test('should render ImageList component', () => {
    const wrapper = mount(<ImageList {...compoentProps} />);
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('should render 2 Image component', () => {
    const wrapper = mount(<ImageList {...compoentProps} />);
    expect(wrapper.find('Image').length).toBe(2);
  });
});