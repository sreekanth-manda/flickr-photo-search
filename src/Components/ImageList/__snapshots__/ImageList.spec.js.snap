// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageList should render ImageList component 1`] = `"<div class=\\"image-list\\"><div class=\\"image-item\\"><img class=\\"image\\" src=\\"https://farm66.staticflickr.com/65535/48458406181_23a99e54c0.jpg\\" alt=\\"\\"><div class=\\"image-metadata\\"><span>Owner: </span><span>Date Taken: </span><span>Click <a class=\\"link\\" target=\\"_blank\\" rel=\\"noopener noreferrer\\">here</a> for full Image</span><div><span>Tags:</span></div></div></div><div class=\\"image-item\\"><img class=\\"image\\" src=\\"https://farm68.staticflickr.com/65535/232323123_23a99e54c0.jpg\\" alt=\\"\\"><div class=\\"image-metadata\\"><span>Owner: </span><span>Date Taken: </span><span>Click <a class=\\"link\\" target=\\"_blank\\" rel=\\"noopener noreferrer\\">here</a> for full Image</span><div><span>Tags:</span></div></div></div></div>"`;
